### PR TITLE
Bluetooth: Audio: Spring cleaning

### PR DESCRIPTION
**Pull request by Thalley**
Adds, removes and modifies includes in all LE audio files.

Fixes any found spelling mistakes as well.

Fixes a few places where incorrect types were used.

### CODE REVIEW

**Comment by Thalley** [comment]
For reviewers: This ended up being much larger than anticipated. Let me know if you want me to split it up into multiple, but similar, PRs.

File: subsys/bluetooth/audio/ascs_internal.h [hunk 1]
```diff
@@ -7,6 +7,13 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifndef __BT_ASCS_INTERNAL_H
```
Comment (wopu-ot) [inline subsys/bluetooth/audio/ascs_internal.h:10]: From the C99 standard:

> All identifiers that begin with an underscore and either an uppercase letter or another underscore are always reserved for any use.

Comment (Thalley) [reply subsys/bluetooth/audio/ascs_internal.h]: Changed. 

We do have **a lot** of macros in Zephyr that does not comply with this though.

**Comment by Thalley** [comment]
@larsgk 

Without PR
Memory region         Used Size  Region Size  %age Used
           FLASH:      948828 B         1 MB     90.49%
             RAM:      405799 B       448 KB     88.46%
        IDT_LIST:          0 GB        32 KB      0.00%

With PR
Memory region         Used Size  Region Size  %age Used
           FLASH:      948852 B         1 MB     90.49%
             RAM:      405799 B       448 KB     88.46%
        IDT_LIST:          0 GB        32 KB      0.00%

There is a small increase in FLASH, but that may due to the few bugfixes or fixed typos in log statements that are in there too.

**Review by kruithofa — DISMISSED** [review]

**Comment by larsgk** [comment]
I would not include autoconf.h as it's generated and available to all code.  Also, IMO (religion perhaps), I would keep the includes section in C files as minimal as possible because developers making changes usually only look in that section if something is missing (doesn't compile).

**Review by kruithofa — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Review by kruithofa — APPROVED** [review]

**Review by pin-zephyr — APPROVED** [review]
LGTM

**Comment by Thalley** [comment]
The tool used to fix the includes was `clangd`

**Pull request merged**